kloppy/helpers.py

```diff
@@ -217,10 +219,20 @@ def write_xml_code_data(dataset: CodeDataset, xml_filename: str):
 
 
 def transform(
-    dataset: DatasetT, to_orientation=None, to_pitch_dimensions=None
-) -> DatasetT:
+    dataset: Dataset,
+    to_orientation=None,
+    to_pitch_dimensions=None,
+    to_coordinate_system=None,
```

Comment:
The `to_coordinate_system` does also accept `Provider`. Think it's better to change the signature of the function and specify the argument types to help us remind what types are accepted.